chain/network/src/peer_manager/testonly.rs

```diff
@@ -0,0 +1,57 @@
+use crate::network_protocol::testonly as data;
+use crate::peer_manager::peer_manager_actor::Event as PME;
+use crate::testonly::actix::ActixSystem;
+use crate::testonly::fake_client;
+use crate::PeerManagerActor;
+use crate::Sink;
+use actix::Actor;
+use near_network_primitives::types::NetworkConfig;
+use near_store::test_utils::create_test_store;
+use std::sync::Arc;
+use tokio::sync::mpsc;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Event {
+    Client(fake_client::Event),
+    PeerManager(crate::peer_manager::peer_manager_actor::Event),
+}
+
+pub struct Handler {
+    pub cfg: NetworkConfig,
+    _actix: ActixSystem<PeerManagerActor>,
+    event_stream: mpsc::UnboundedReceiver<Event>,
+}
+
+impl Handler {
+    pub async fn recv(&mut self) -> Event {
+        self.event_stream.recv().await.unwrap()
+    }
+    // consume events, until Event::PeerManager happens.
+    pub async fn recv_peer_manager(&mut self) -> crate::peer_manager::peer_manager_actor::Event {
+        loop {
+            if let Event::PeerManager(ev) = self.recv().await {
```

Comment:
A bit odd that we drop all other messages on the floor here, I'd probably expect some buffering here: 

https://github.com/rust-lang/rust-analyzer/blob/32b40ded0f16818918d4a2ca16400526eea369f2/crates/rust-analyzer/tests/slow-tests/support.rs#L259-L285

But if it works, it works

Reply:
So there are 2 cases that I'd like to support for testing:
- we are waiting for some specific event, and expect it to be the first one (modulo some filter, ignoring the rest).
- we are aggregating all the events (in some way), waiting for the state to satisfy some predicate.

Best I can do here (I think) is to rewrite these methods to be sth like "fn consume_until<T>(filter:Event->Option<T>) -> T", so that it is explicit that this is a helper for the first use case.

Reply:
Nvm, buffering the filtered out messages also should work.

Reply:
So to achieve non-dropping stream of events we need an unbounded version of tokio::sync::broadcast. I'll implement one.